lib/galaxy/datatypes/genetics.py

```diff
@@ -827,6 +829,297 @@ def __init__(self, **kwd):
                                 substitute_name_with_metadata='base_name', is_binary=True)
 
 
+class LinkageStudies(Text):
+    """
+    superclass for classical linkage analysis suites
+    """
+    test_files = [
+        'linkstudies.allegro_fparam', 'linkstudies.alohomora_gts',
+        'linkstudies.linkage_datain', 'linkstudies.linkage_map'
+    ]
+
+    def __init__(self, **kwd):
+        Text.__init__(self, **kwd)
+        self.lcount = 0
+        self.max_lines = 10
+        # iterated whole file without errors
+        self.eof_res = True
```

Comment:
I'm not a big fan of this. Compared to most other datatypes I've read / written which just return true if they want to return true rather than returning a class variable.